pkg/manifests/manifests_test.go

```diff
@@ -4256,14 +4259,16 @@ func TestPrometheusGoGC(t *testing.T) {
 				}
 			}
 			require.NotNil(t, c)
-			if tc.exp == "" {
+			if tc.expectedGOGC == "" {
 				for _, env := range c.Env {
 					require.NotEqual(t, env.Name, "GOGC")
 				}
 				return
 			}
 
-			require.Contains(t, c.Env, v1.EnvVar{Name: "GOGC", Value: tc.exp})
+			require.Contains(t, c.Env, v1.EnvVar{Name: "GOGC", Value: tc.expectedGOGC})
+
+			require.Equal(t, tc.autoGOMEMLIMITDisabled, argumentPresent(*c, "--no-auto-gomemlimit"))
```

Comment:
We could drop the `tc.autoGOMEMLIMITDisabled` field as this could be safely derived from `ir.HighlyAvailableInfrastructure()`, as that's the only case where this is disabled for now (else enabled)?
```suggestion
			require.Equal(t, tc.ir.HighlyAvailableInfrastructure(), argumentPresent(*c, "--no-auto-gomemlimit"))
```

Reply:
yep, but I'm trying to avoid the same logic in tests as well in case sth is broken/wrong somewhere.
Also having it explicit makes reading the test cases easier.